controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -775,11 +787,34 @@ public CompletableFuture<CreateStreamStatus.Status> createStream(String scope, S
                                                                      long createTimestamp, long requestId) {
         log.debug(requestId, "createStream with resource called.");
         OperationContext context = streamMetadataStore.createStreamContext(scope, stream, requestId);
-
-            return execute(
+        if (config.getRetentionPolicy() == null && this.globalRetentionPolicy.get()) {
+            config = createRetentionPolicy(config);
+        }
+        final StreamConfiguration  streamConfig = config;
+        return execute(
                     new Resource(scope, stream),
-                    new Serializable[]{scope, stream, config, createTimestamp, requestId},
-                    () -> createStreamBody(scope, stream, config, createTimestamp, context));
+                    new Serializable[]{scope, stream, streamConfig, createTimestamp, requestId},
+                    () -> createStreamBody(scope, stream, streamConfig, createTimestamp, context));
+    }
+
+    private StreamConfiguration createRetentionPolicy(StreamConfiguration config) {
+        Preconditions.checkArgument(this.minRetentionValue.get() > 0, "Min retention value must be > 0.");
+        RetentionPolicy retentionPolicy = null;
+        if (this.retentionType.get().equalsIgnoreCase("time")) {
```

Comment:
What is this hardcoded "time" string here? How are the original retention policies being parsed? Do they use the same approach or there is a better way to distinguish between retention policy types¿? I'm sure this validation happens somewhere else and can be reused.

Reply:
Till now the retention policies are set while creating `StreamConfiguration`. That time there is an option to choose the type by selecting correct `enum` type. Here we are trying to set the type based on the values received via Config param

Reply:
Property with enum might be better choice in this case. See example here https://github.com/pravega/pravega/blob/2cd0c8c1521e21844435f0bb18311d9b030fbe2b/segmentstore/server/src/main/java/io/pravega/segmentstore/server/store/ServiceConfig.java#L73

Reply:
done, using `Enum` now